### PR TITLE
Fix for symlink support.

### PR DESCRIPTION
**Pull request by nitin-deamon**

### CODE REVIEW

File: cmd/zc_traverser_local.go [hunk 1]
```diff
@@ -234,8 +253,9 @@ func WalkWithSymlinks(appCtx context.Context, fullPath string, walkFunc filepath
 	// do NOT put fullPath: true into the map at this time, because we want to match the semantics of filepath.Walk, where the walkfunc is called for the root
 	// When following symlinks, our current implementation tracks folders and files.  Which may consume GB's of RAM when there are 10s of millions of files.
 	var seenPaths seenPathsRecorder = &nullSeenPathsRecorder{} // uses no RAM
+	var haveSeenPaths = &haveSeenSymlinkPath{}
```
Comment (linuxsmiths) [inline cmd/zc_traverser_local.go:256]: nit, let's call haveSeenPaths -> seenSymlinkTargets
for better readability

**Review by linuxsmiths — COMMENTED** [review]
let's discuss these

File: cmd/zc_traverser_local.go [hunk 2]
```diff
@@ -184,6 +184,10 @@ type realSeenPathsRecorder struct {
 	m map[string]struct{}
 }
 
+type haveSeenSymlinkPath struct {
```
[the remark this answers is not in the document]
Comment (nitinsingla81) [reply cmd/zc_traverser_local.go:187]: done, kept the realSeenPath as of now, delete once ze reply back.

File: cmd/zc_traverser_local.go [hunk 3]
```diff
@@ -192,6 +196,21 @@ func (r *realSeenPathsRecorder) HasSeen(path string) bool {
 	return ok
 }
 
+func (r *haveSeenSymlinkPath) Record(symPath, path string) {
```
[the remark this answers is not in the document]
Comment (nitinsingla81) [reply cmd/zc_traverser_local.go:199]: done

File: cmd/zc_traverser_local.go (hunk 1 above)
Comment (nitinsingla81) [reply cmd/zc_traverser_local.go:256]: done

**Review by linuxsmiths — APPROVED** [review]

**Pull request merged**